dev_process.md

```diff
@@ -0,0 +1,86 @@
+# Development Process
+One of the most crucial factors in having a secure codebase is a solid development process. "An ounce of prevention is worth a pound of cure". This document gives an *example* development process that we, at Nascent, have found works well. In addition to getting better overall code quality, this process serves as a companion to our [audit readiness checklist](https://github.com/nascentxyz/nascent-security/blob/main/audit-readiness-checklist.md). If you follow the below process, you should naturally check off most of the items on the checklist.
+
+```
+Feature Request received
+ |
+ └> Specification Write up
+     |
+     └> Evaluation (Time + complexity + risks)
+        |
+        └> Implementation
+           |
+           └> Test
+              |
+              └> Deployment
+                 |
+                 └> Monitor
+```
+## Specification
+Make note of the kinds of variables that affect the feature:
+1. User input?
+2. Time?
+3. Other protocols?
+4. Existing state?
+
+## Note on Evaluation
+Make dedicated time to evaluate:
+1. How long will this realistically take? Without specification most estimates will be wrong.
+2. Is the specification overly complex? Complexity leads to bugs and worse overall code
+3. What are the risks associated with this feature? Spend ample time evaluating this. Consider every module the feature may touch. Then go back through excluded modules and *ensure* they cannot be affected.
+
+Now that you have millions of dollars at risk, you have to consider each feature may put *all* of it at risk. The cost of a misstep will likely be in the millions of dollars. 
+## Implementation -> Monitor
+1. Draft PR
+  - Include the feature request + specification
+2. Initial implementation pass
+  - Get an initial implementation in place
+  - Document all functions' intended behavior ([NatSpec](https://docs.soliditylang.org/en/develop/natspec-format.html) for public/external functions) and add inline documentation/comments
+  - Any use of `unchecked` should include `safety` documentation like:
+    ```solidity
+    // Safety:
+    //  1. a + b: a and b are uint64s, and a is casted up to uint128, so a uint128(uint64) + uint64
+    //     cannot overflow
+    //  2. c * 2: c is casted up to uint256, so a uint256(uint128) * 2 cannot overflow
+    uint256 d;
+    unchecked {
+      uint128 c = uint128(a) + b;
+      d = uint256(c) * 2;
+    }
+    ```
+  - **Every** line of assembly is documented/commented on
+3. Initial concrete [tests](https://book.getfoundry.sh/forge/tests.html)
```

Comment:
Is there a good way to pull out 3-5 as a section or subsection while retaining the numbering?

Reply:
Actually, this seems like it goes through 10 or 11. Any reason not to just pull out Testing as its own section?

Reply:
Will do